src/app/public/modules/checkbox/checkbox.component.ts

```diff
@@ -175,6 +178,17 @@ export class SkyCheckboxComponent implements ControlValueAccessor, Validator {
   public validate(control: AbstractControl): ValidationErrors {
     if (!this.control) {
       this.control = control;
+
+      // Get required state from AbstractControl.
+      // https://github.com/angular/components/issues/2574#issuecomment-461185408
+      const vf = this.control.validator(new FormControl());
+      this.required = vf ? vf.required : undefined;
+    }
+
+    if (this.required && !control.value) {
+      return {
+        'required': true
```

Comment:
If I check the checkbox it becomes "valid", but if I uncheck it, it remains "valid". Shouldn't it be invalid?

Reply:
Yes, but I'm unable to replicate this. Are you doing this with the visual demo, or did you make your own component?

Reply:
I encountered this on the visual demo.